src/Microsoft.ML.Data/DataLoadSave/Text/TextLoader.cs

```diff
@@ -1473,12 +1490,14 @@ void ICanSaveModel.Save(ModelSaveContext ctx)
            bool allowQuoting = Defaults.AllowQuoting,
            bool supportSparse = Defaults.AllowSparse,
            bool trimWhitespace = Defaults.TrimWhitespace,
-           IMultiStreamSource dataSample = null)
+           IMultiStreamSource dataSample = null,
+           char decimalMarker = Defaults.DecimalMarker)
         {
```

Comment:
Here I placed `decimalMarker` after `dataSample` instead of after `seperator` to avoid editing constructors in TextLoaderSaverCatalog, which results in API breaking changes.

Reply:
So I had commented to @harishsk that maybe it's better to not add new parameters to this internal method, precisely because it's used by the TextLoaderSaverCatalog's API, and we can't change that other public method. That's why I didn't add the readMultiline option I added on #5125 in this method.

Originally I thought suggesting him that we should deprecate the methods that have a list of parameters, and only leave those which use an Options object, because that way we won't need to worry about API breaking changes... but now I'm not so sure about that, and so I didn't suggest it. So perhaps it would be nice to know if you have any opinion on this.

Reply:
In my opinion I agree that we should not continue to add new parameters to this internal method as we can't change `TextLoaderSaverCatalog` due to it being an API breaking change. We should definitely prefer methods that use an Options object instead of those with a list of parameters. As such I'm removing this addition, and am in support of marking these parameter-listed-methods to be marked with `ObsoleteAttribute`.

---
In reply to: [427823255](https://github.com/dotnet/machinelearning/pull/5145#discussion_r427823255) [](ancestors = 427823255)